File: quick-test.js
// quick-test.js - Quick verification for Railway deployment
import fetch from 'node-fetch';

async function quickTest() {
  console.log('🏀 NBA Backend Railway Deployment Test\n');
  
  const criticalEndpoints = [
    '/health',
    '/api/health',
    '/api/nba',
    '/api/players',
    '/api/teams',
    '/api/auth'
  ];
  
  console.log('Testing critical endpoints...\n');
  
  for (const endpoint of criticalEndpoints) {
    try {
      const response = await fetch(`http://localhost:3002${endpoint}`);
      const status = response.status;
      
      if (status >= 200 && status < 300) {
        console.log(`✅ ${endpoint} - HTTP ${status}`);
      } else {
        console.log(`❌ ${endpoint} - HTTP ${status}`);
      }
    } catch (error) {
      console.log(`❌ ${endpoint} - ${error.message}`);
    }
  }
  
  console.log('\n✅ Quick test complete!');
  console.log('If all endpoints return 200, backend is ready for Railway.');
  console.log('\n🚀 To deploy to Railway:');
  console.log('1. Commit all changes: git add . && git commit -m "Ready for Railway"');
  console.log('2. Push to Railway: git push railway main');
}

// Run the test
quickTest().catch(console.error);

File: quick_test.py
import requests
import json

def test_api(port=3002):
    base_url = f"http://localhost:{port}"
    
    print(f"Testing API on port {port}...\n")
    
    endpoints = [
        "/api/health",
        "/api/players",
        "/api/fantasy/teams"
    ]
    
    for endpoint in endpoints:
        url = base_url + endpoint
        print(f"Testing {endpoint}...")
        try:
            response = requests.get(url, timeout=5)
            if response.status_code == 200:
                data = response.json()
                print(f"  ✅ Status: {response.status_code}")
                if endpoint == "/api/players":
                    print(f"  📊 Players: {data.get('count', 0)}")
                elif endpoint == "/api/fantasy/teams":
                    print(f"  🏀 Teams: {data.get('count', 0)}")
                else:
                    print(f"  📍 Response: {data.get('status', 'OK')}")
            else:
                print(f"  ❌ Status: {response.status_code}")
        except Exception as e:
            print(f"  ❌ Error: {e}")
        print()

if __name__ == "__main__":
    # Try multiple ports
    ports_to_try = [3002, 5000, 5001, 8888]
    
    for port in ports_to_try:
        print(f"=" * 40)
        print(f"Trying port {port}")
        print(f"=" * 40)
        test_api(port)
